bigtable/google/cloud/bigtable/mutation.py

```diff
@@ -0,0 +1,125 @@
+
+        Arguments:
+            row_key (str): Key of the Row in string.
+    """
+    ALL_COLUMNS = object()
+    """Sentinel value used to indicate all columns in a column family."""
+
+    def __init__(self, row_key, app_profile_id=None):
+        self.row_key = _to_bytes(row_key)
+        self.app_profile_id = app_profile_id
+        self.mutations = []
+
+    def set_cell(self, family_name, column_id, value, timestamp=None):
+        """Create the mutation request message for SetCell and add it to the
+            list of mutations
+
+        Arguments:
+            family_name (str):
+                The name of the family into which new data should be written.
+                Must match ``[-_.a-zA-Z0-9]+``.
+            column_id (bytes):
+                The qualifier of the column into which new data should be
+                written. Can be any byte string, including the empty string.
+            value (bytes):
+                The value to be written into the specified cell.
+            timestamp (datetime.datetime):
+                (optional) The timestamp of the cell into which new data should
+                be written. Use -1 for current Bigtable server time. Otherwise,
+                the client should set this value itself, noting that the
+                default value is a timestamp of zero if the field is left
+                unspecified. Values must match the granularity of the table
+                (e.g. micros, millis).
+        """
+        if timestamp is None:
+            # Use -1 for current Bigtable server time.
+            timestamp_micros = -1
+        else:
+            timestamp_micros = _microseconds_from_datetime(timestamp)
+            # Truncate to millisecond granularity.
+            timestamp_micros -= (timestamp_micros % 1000)
+
+        set_cell_mutation = data_v2_pb2.Mutation.SetCell(
+            family_name=family_name,
+            column_qualifier=_to_bytes(column_id),
+            timestamp_micros=timestamp_micros,
+            value=value
+        )
+        mutation_message = data_v2_pb2.Mutation(set_cell=set_cell_mutation)
+        self.mutations.append(mutation_message)
+
+    def delete_cells(self, family_name, columns, time_range=None):
+        """Create the mutation request message for DeleteFromColumn and
+            add it to the list of mutations
+
+        Arguments:
+            family_name (str):
+                The name of the family into which new data should be written.
+                Must match ``[-_.a-zA-Z0-9]+``.
+            columns (list):
+                The columns within the column family that will have cells
+                deleted. If :attr:`ALL_COLUMNS` is used then the entire
+                column family will be deleted from the row.
+            time_range (TimestampRange):
+                (optional) The range of timestamps within which cells should be
+                deleted.
+        """
+        if columns is self.ALL_COLUMNS:
```

Comment:
I don't think that ALL_COLUMNS is a good idea.  Users should just call `delete_from_family` directly